### DISCUSSION TITLE
Can't get web interface to work

### DISCUSSION THREAD

**Discussion by buffeh** [judahpaul16/gpt-home#37] [Q&A] [answered]
Hi, hope someone can help. 

I didn't think I'd have any issues as it uses a setup script to basically sort everything for you. So I thought even my self with some but not a lot of Linux knowledge would be fine! I even made sure to use Ubuntu Server and a Raspberry Pi 4 - just like the developer - to reduce the chance of issues. 

The setup script didn't work 100% like the Github page said it would, and would fail, until I installed the dependencies manually then it all worked beautifully and the services etc were setup.

The project is useless without the web interface, and as far as I could tell, this isn't set up by default, and the instructions at https://github.com/judahpaul16/gpt-home/tree/main/src/frontend need to be followed to get it up and running and hosted.

So - I followed these instructions exactly and ran sudo npm install (I had to actually install npm first), and that ran ok, with a few warnings about depreciated things but no critical errors. Then I ran sudo npm start, and it compiled and started hosting on port 3000 with no errors, it specifically said all is ok.

I was then able to connect and see the web page, the Integrations page seemed to work fine, but the event logs page would show an error "JSON.parse: unexpected character at line 1 column 1 of the JSON data" and the settings page would just say "Loading" forever.

Additionally if I try and build the project with sudo npm run build, that works ok, but when I serve it with sudo serve -s build, it starts serving fine, but when looking at the web page at localhost:3000 I either get just a directory listing of the files in the frontend/build folder, or I get a screen prompting me to choose a password, but if I try and set one, I get the error "Error hashing password: undefined". I would expect to at least see the same website as when running sudo npm test, even if it's broken.

I have absolutely no idea what I could have done wrong or what the problem is. I would imagine if everyone was having this problem, it would be apparent in the Issues section on Github, so I don't think it's a problem with the project in general.

If anyone could offer any advice or point me in the right direction to try and get this sorted, I'd be really grateful.

**Comment by buffeh** [comment, marked as answer]
Fixed it, I was mistaken about having to setup the Frontend web page manually, the setup script did do it and the web interface is actually on port 80 (not 3000) - for anyone else who finds this :)

**Comment by judahpaul16** [comment]
You got it. You can see the status of all docker services by running `docker exec -it gpt-home supervisorctl status`. The instructions at https://github.com/judahpaul16/gpt-home/tree/main/src/frontend are for if you want to make changes to the web interface. When you run `npm install` and `npm run start` it will launch a development server ***on the computer you run the commands on*** at `localhost:3000` so you can see the frontend changes you make. You would then do `npm run build` to rebuild the frontend to the `build` folder and run `docker exec -it gpt-home restart web-interface` to have the changes reflected in the docker container.
